deploy/crds/devconsole_v1alpha1_gitsource_crd.yaml

```diff
@@ -9,38 +9,67 @@ spec:
     listKind: GitSourceList
     plural: gitsources
     singular: gitsource
-    shortNames:
-      - gitsrc
   scope: Namespaced
+  subresources:
+    status: {}
   validation:
-    # openAPIV3Schema is the schema for validating custom objects.
     openAPIV3Schema:
       properties:
         apiVersion:
+          description: 'APIVersion defines the versioned schema of this representation
+            of an object. Servers should convert recognized schemas to the latest
+            internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#resources'
           type: string
         kind:
+          description: 'Kind is a string value representing the REST resource this
+            object represents. Servers may infer this from the endpoint the client
+            submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds'
           type: string
         metadata:
           type: object
         spec:
           properties:
-            url:
-              type: string
-            ref:
-              type: string
             contextDir:
+              description: ContextDir is a path to subfolder in the repo. Optional.
+              type: string
+            flavor:
+              description: Flavor of the git provider like github, gitlab, bitbucket,
+                generic, etc. Optional.
               type: string
             httpProxy:
+              description: HttpProxy is optional.
               type: string
             httpsProxy:
+              description: HttpsProxy is optional.
               type: string
             noProxy:
+              description: NoProxy can be used to specify domains for which no proxying
+                should be performed. Optional.
+              type: string
+            ref:
+              description: Ref is a git reference. Optional. "master" is used by default.
               type: string
             secretRef:
+              description: SecretRef refers to the secret that contains credentials
+                to access the git repo. Optional.
               properties:
                 name:
+                  description: Name is the name of the secret that contains credentials
+                    to access the git repo
```

Comment:
Not to be a jerk - can we be consistent and put periods at the end of all these descriptions?